### ISSUE TITLE
Conditional keys added to Partial<T> are not checked

### ISSUE THREAD

**Issue by pierpo** [microsoft/TypeScript#38896]
**Search Terms:**

conditional keys object properties not checked

**Code**

```ts
interface MyInterface {
    a: number;
    b: number;
}

const myFunc = (): Partial<MyInterface> => {
    return {
        a: 1,
        b: 2,
        ...(true && { c: 'oops, no error' }),
    }
}
```

**Expected behavior:**

I should have an error saying "`c` key does not exist in `MyInterface`".

**Actual behavior:**

I do not get an error.

**Playground Link:** 

[Playground Link](https://www.typescriptlang.org/play/?ssl=12&ssc=2&pln=1&pc=1#code/JYOwLgpgTgZghgYwgAgLIE8CS5rycgbwChlTk4AuZEAVwFsAjaAbhLIatsZaIF8iiCAPYgAzmGR10AMRogEyALzIAFAEoqABThQwwOABsAPBmyRYiCAD4lN4mWRQIYGlBCE2D0pWQBGADSeXhzIAEyBXmQAdDEqYFA0KABkSYTICFQA5EJCAA6i-tRCyNBQQlCZyLxqEWT8vEA)

**Related Issues:** 

I could not find any.

____

Please note that the values are checked. This does produce an error:

```ts
interface MyInterface {
    a: number;
    b: number;
}

const myFunc = (): Partial<MyInterface> => {
    return {
        a: 1,
        ...(true && { b: 'okay I get an error' }),
    }
}
```

**Comment by MartinJohns** [comment]
Excess property checks do not occur for spread objects. This is intentional, see #19775.

**Comment by pierpo** [comment]
Oh! Indeed. I did not pinpoint the problem precisely enough, I should've known it was related to spread operators.

Thank you for your very quick and helpful answer! You may close the issue! Sorry about the duplicate :blush:
